### PR TITLE
feat(transform): added transform plugin , runtime controller, React hook 

### PR DESCRIPTION
**Pull request by makhnatkin**
Added first version html plugin
1. plugin/transform.ts

```
export type PluginOptions = {
    runtimeJsPath: string;
    containerClasses: string;
    bundle: boolean;
    sanitize?: (dirtyHtml: string) => string;
};
```

2. runtime/HtmlController.ts
```
export interface IHtmlIFrameController {
    readonly block: HTMLIFrameElement;
    execute(callback: (controller: IHtmlIFrameController) => void): void;
    setStyles(styles: Record<string, string>): void;
}
```
`HtmlIFrameController` class has also been added that can automatically adjust the height of an iframe based on its content, apply styles, and execute callbacks for each of the frames.

3. react/useDiplodocHtml.ts with methods `reinitialize`, `forEach`
4. Tests were not added in this PR, they will be implemented later.

### CODE REVIEW

File: src/runtime/utils.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+export const setIframeStyles = (block: HTMLIFrameElement, styles: Record<string, string>) => {
+    Object.keys(styles).forEach((property) => {
+        block.contentWindow?.document.body.style.setProperty(property, styles[property]);
+    });
+};
+
+// @ts-ignore  @typescript-eslint/no-explicit-any
+type QueueManagerCallback<T> = (...args: T[]) => any | void;
+
+// @ts-ignore  @typescript-eslint/no-explicit-any
```
Comment (d3m1d0v) [inline src/runtime/utils.ts:19]: why is `@ts-ignore` used?

File: src/runtime/HtmlController.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    isHTMLElement,
+    isIFrameLoaded,
+    resizeIframeToFitContent,
+    setIframeStyles,
+} from './utils';
+
+const IFRAME_TAG = 'iframe';
+
+export type HTMLControllerForEachCallback = (
+    block: HtmlDivController | HtmlIFrameController,
```
Comment (d3m1d0v) [inline src/runtime/HtmlController.ts:14]: maybe it's worth adding an interface that will be implemented by  HtmlDivController and HtmlIFrameController?

File: README.md [hunk 3]
```diff
@@ -1,11 +1,148 @@
...
+    <h2>14 June 2024</h2>
+    <p>Heavy rain.</p>
+  </article>
+</article>
+
+:::
+`, {
+    plugins: [
+        htmlExtension.transform({
+            sanitize: dirtyHtml => sanitizeHtml(dirty, {
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply README.md]: fixed

File: package.json [hunk 4]
```diff
@@ -1,32 +1,86 @@
...
+  "main": "plugin/index.js",
+  "types": "plugin/index.d.ts",
+  "exports": {
+    ".": {
+      "types": "./plugin/index.d.ts",
+      "default": "./plugin/index.js"
+    },
+    "./runtime": {
+      "types": "./runtime/index.d.ts",
+      "style": "./runtime/index.css",
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply package.json]: deleted css (it was empty)

File: src/plugin/transform.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+const emptyOptions = {};
+
+export function transform({
+    runtimeJsPath = '_assets/html-extension.js',
+    runtimeCssPath = '_assets/html-extension.css',
+    containerClasses = '',
+    bundle = true,
+    sanitize = defaultSanitize,
+    shouldUseSanitize = false,
+    shouldUseIframe = true,
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply src/plugin/transform.ts]: deleted

File: src/plugin/transform.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+const emptyOptions = {};
+
+export function transform({
+    runtimeJsPath = '_assets/html-extension.js',
+    runtimeCssPath = '_assets/html-extension.css',
+    containerClasses = '',
+    bundle = true,
+    sanitize = defaultSanitize,
+    shouldUseSanitize = false,
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply src/plugin/transform.ts]: fixed, now only: transform({sanitize: (v) => v}); or transform({sanitize: undefined});

File: src/plugin/transform.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+            if (shouldUseIframe) {
+                token.attrPush([TokenAttr.dataId, htmlBlockId]);
+                token.attrPush([TokenAttr.dataKey, BLOCK_NAME]);
+                token.attrPush([TokenAttr.frameborder, '0']);
+                token.attrPush([TokenAttr.id, htmlBlockId]);
+                token.attrPush([TokenAttr.style, 'width:100%']);
+            }
+
+            const resultHtml = shouldUseSanitize ? sanitize(content) : content;
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply src/plugin/transform.ts]: moved to render

File: src/plugin/utils.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import sanitizeHtml from 'sanitize-html';
```
[the remark this answers is not in the document]
Comment (makhnatkin) [reply src/plugin/utils.ts]: deleted default sanitizeHtml

File: src/runtime/HtmlController.ts (hunk 2 above)
Comment (makhnatkin) [reply src/runtime/HtmlController.ts]: deleted HtmlDivController

**Review by d3m1d0v — APPROVED** [review]

**Pull request merged**